Tests/Pcap++Test/Tests/FileTests.cpp

```diff
@@ -974,3 +974,44 @@ PTF_TEST_CASE(TestSolarisSnoopFileRead)
 
 	readerDev.close();
 }  // TestSolarisSnoopFileRead
+
+PTF_TEST_CASE(TestPcapFileWriterDeviceDestructor)
```

Comment:
It's a bit hard to read and understand your intention. Please simplify the test, e.g. reduce the writes, and add more comments.